src/openforms/forms/admin/tasks.py

```diff
@@ -102,3 +104,40 @@ def process_forms_import(import_file: str, user_id: int) -> None:
 def clear_forms_export():
     logger.debug("Clearing old export files")
     call_command("delete_export_files")
+
+
+@app.task()
+def activate_forms():
+    """Activate all the forms that should be activated by the specific date and time."""
+    now = timezone.now()
+    forms = Form.objects.filter(
+        active=False,
+        _is_deleted=False,
+        activate_on__lte=now,
+        activate_on__gt=now - timedelta(minutes=5),
+    )
+
+    for form in forms:
+        form.active = True
+        form.activate_on = None
+        logger.debug(f"Activated form {form.admin_name}")
+        logevent.form_activated(form)
+
+    Form.objects.bulk_update(forms, fields=["active", "activate_on"])
```

Comment:
```suggestion
    for form in forms:
        form.active = True
        form.activate_on = None
        logger.debug("Activating form %s", form.admin_name)
        transaction.on_commit(lambda: logevent.form_activated(form))

    Form.objects.bulk_update(forms, fields=["active", "activate_on"])
```
& then also decorate the task with `@transaction.atomic`

So, a couple things here:

1. The logging - never do the interpolation directly in the log message, since Sentry aggregates logs by their message template. My code suggestion ensures that there's only a single Sentry record for this activation, while having many events each with their own context and timestamp for every form.
2. the transactions - it's now possible (with autocommit, the default) that a log event is written claiming activation, but the actual `bulk_update` query failing. So now we have a lying log record, because the form failed to activate.

I don't expect failures now, but I'm also not ruling them out, weird things can happen.

I don't really like wrapping the entire task in a transaction.atomic either (as I suggest) - one broken form would then possibly prevent other forms from being activated. So probably the most robust way to handle this is to activate each form individually and wrap each one in its own transaction:

```py
for form in forms:
    form.active = True
    form.activate_on = None
    with transaction.atomic():
        form.save(update_fields=["active", "activate_on"])
        logevent.form_activated(form)
```

I would then make that even nicer by putting all that activation logic in a model method, so your task becomes:

```py
for form in forms:
    with transaction.atomic():
         try:
            form.activate()
        except DatabaseError as exc:
            logger.error("Form activation of form %s failed", form.admin_name, exc_info=exc, extra={"pk": form.pk})
```

and then implement `def activate` to set the appropriate fields + write to the event log.